### DISCUSSION TITLE
Adaptive font size

### DISCUSSION THREAD

**Discussion by tbmuc** [ImageMagick/ImageMagick#2115] [Help]
Hello,

I' looking for a method to fit text into a box size restrict it to 100pt font size at the same time.
That means, if the text fits into the box, the font size should be 100pt, if not it should be lower. However it should not be larger than 100pt in any case.

**Comment by snibgo** [comment]
You can find what pointsize IM would use, eg:

```
magick -size 400x300 caption:"Hello world" -format %[caption:pointsize] info:
132
```
In a shell script you can test this number for being larger than 100. Then use your required pointsize in your final command.

**Comment by tbmuc** [comment]
Is there a way to do that automatically without user interaction depending on the text?. It should be use for mass generation of images.

**Reply by fmw42** [reply]
It can be scripted to test and use whichever is appropriate.  What is your IM version and platform?

**Comment by snibgo** [comment]
Yes, with a shell script.

**Comment by tbmuc** [comment]
I mean calculating all values without feedback of the user. The user shall only enter the text and get the jpg. The pointsize should not be given by the user. 

In the docu are several examples where image magic makes a text fit into a box. What I'm looking for is some parameter like "maxfontsize", which restricts the font size.

**Comment by snibgo** [comment]
Calculating all values? I don't understand. You are specifying the box width and height, aren't you?

**Comment by tbmuc** [comment]
Yes, box size is fix. The text may fit in one row or two. If it fit in 2 rows the fontsize may lower than 100. 

Lets have 2 examples, maximum font size shall be 100. Box may be 200x200.

a) Text is just an A: Font size should be 100px, even if a 200px A would fit into the box
b) Text is very long Like "how much wood would w woodchuck chuck": Now, the font size may be eg 10px to make the text fit into the box.

The font size calculation shall be done automatically without interaction of the user.

**Comment by snibgo** [comment]
If you need help writing the shell script, you should say what shell you use (bash, Windows BAT, or whatever). Also say what version of IM you use.

**Comment by tbmuc** [comment]
Windows 10 64 Bit, BAT,  Imagemagic 7.0.10
The script shall have only the text as parameter.

**Comment by snibgo** [comment]
```
rem %1 is quoted string.

setlocal enabledelayedexpansion

for /F "usebackq" %%A in (`%IMG7%magick ^
  -size 200x200 ^
  caption:%1 ^
  -format %%[caption:pointsize] ^
  info:`) do (
  set PNTSZ=%%A
)

if %PNTSZ% GTR 100 set PNTSZ=100

%IMG7%magick ^
  -size 200x200 ^
  -pointsize %PNTSZ% ^
  caption:%1 ^
  -format %%[caption:pointsize] ^
  out.png
```

**Comment by tbmuc** [comment]
Great. Thank you very much.
